### DISCUSSION TITLE
Setting Configuration Parameters for a Role

### DISCUSSION THREAD

**Discussion by jcarvalho** [cyrilgdn/terraform-provider-postgresql#176] [Q&A]
Hi.

We're looking to set per-role configuration parameters (as specified in https://www.postgresql.org/docs/11/sql-alterrole.html).

Is there a native way to do this with the provider? Couldn't find anything about it so far.

**Comment by wimi** [comment]
Hi there, a few of role parameters can be setup using properties of role resource directly, but general approach to specify any parameter per role is not natively available.
I've started similar discussion some time ago (https://github.com/cyrilgdn/terraform-provider-postgresql/discussions/86) and even had some POC code implementing that (links in discussion), but it seems that generally it's not widely looked for feature.

At the end we've implemented our own dynamic provider in Pulumi to workaround this.
